### PR TITLE
Add back TI OTA PAL port for v3 of the OTA library

### PR DESCRIPTION
**Pull request by yanjos-dev**
Add back TI OTA PAL port for v3 of the OTA library

Description
-----------
* Add OTA support to the TI CMakeLists.txt file.
* Add OTA related config files to the demo and test projects.
* Add the OTA PAL port files ota_pal.c and ota_pal.h.
* Update the TI IAR and CCS demo and test project files for OTA.

Note:
This PR is taking a PR that was merged into a development branch, and then applying those changes to mainline. The original PR along with its review can be found in PR https://github.com/aws/amazon-freertos/pull/3179.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vendors/ti/boards/cc3220_launchpad/CMakeLists.txt [hunk 1]
```diff
@@ -163,6 +163,33 @@ target_sources(
     INTERFACE "${afr_ports_dir}/secure_sockets/iot_secure_sockets.c"
 )
 
+# OTA
```
Comment (divekarshubham) [inline vendors/ti/boards/cc3220_launchpad/CMakeLists.txt:166]:
```suggestion
# Over-the-air Updates
```

File: vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_demo_config.h [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+/**
+ * @brief Server's root CA certificate.
+ *
+ * This certificate is used to identify the AWS IoT server and is publicly available.
+ * Refer to the AWS documentation available in the link below for information about the
+ * Server Root CAs.
+ * https://docs.aws.amazon.com/iot/latest/developerguide/server-authentication.html#server-authentication-certs
+ *
+ * @note The TI C3220 Launchpad board requires that the Root CA have its certificate self-signed. As mentioned in the
```
Comment (divekarshubham) [inline vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_demo_config.h:76]: NIT: Could you update the comments to keep it under 80 chars.

File: vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/aws_test_ota_config.h [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+/**
+ * @brief Path to cert for OTA test PAL. Used to verify signature.
+ * If applicable, the device must be pre-provisioned with this certificate. Please see
+ * test/common/ota/test_files for the set of certificates.
+ */
+#define otatestpalCERTIFICATE_FILE                         "rsa-sha1-signer.crt.der"
+
+/**
+ * @brief Some boards have a hard-coded name for the firmware image to boot.
```
Comment (divekarshubham) [inline vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/aws_test_ota_config.h:41]: NIT: should we be specific here?
```suggestion
 * @brief TI CC3220 has a hard-coded name for the firmware image to boot.
```

File: vendors/ti/boards/cc3220_launchpad/ports/ota_pal_for_aws/ota_pal.c [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/* OTA PAL implementation for TI CC3220SF platform. */
+
+#include "ota.h"
+#include "ota_pal.h"
+
+#define kOTA_HalfSecondDelay    pdMS_TO_TICKS( 500UL )
```
Comment (divekarshubham) [inline vendors/ti/boards/cc3220_launchpad/ports/ota_pal_for_aws/ota_pal.c:31]:
```suggestion
#define OTA_HALF_SECOND_DELAY    pdMS_TO_TICKS( 500UL )
```

File: vendors/ti/boards/cc3220_launchpad/CMakeLists.txt (hunk 1 above)
Comment (yanjos-dev) [reply vendors/ti/boards/cc3220_launchpad/CMakeLists.txt]: Done

File: vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_demo_config.h (hunk 2 above)
Comment (yanjos-dev) [reply vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_demo_config.h]: Done

File: vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/aws_test_ota_config.h (hunk 3 above)
Comment (yanjos-dev) [reply vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/aws_test_ota_config.h]: Done

File: vendors/ti/boards/cc3220_launchpad/ports/ota_pal_for_aws/ota_pal.c (hunk 4 above)
Comment (yanjos-dev) [reply vendors/ti/boards/cc3220_launchpad/ports/ota_pal_for_aws/ota_pal.c]: Done

**Review by divekarshubham — DISMISSED** [review]

File: vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_config.h [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+/**
+ * @brief The maximum allowed length of the thing name used by the OTA agent.
+ *
+ * AWS IoT requires Thing names to be unique for each device that connects to the broker.
+ * Likewise, the OTA agent requires the developer to construct and pass in the Thing name when
+ * initializing the OTA agent. The agent uses this size to allocate static storage for the
+ * Thing name used in all OTA base topics. Namely $aws/things/<thingName>
+ */
+#define otaconfigMAX_THINGNAME_LEN             64U
```
Comment (pvyawaha) [inline vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_config.h:95]: Please set this to 128

File: vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/ota_config.h [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+/**
+ * @brief The maximum allowed length of the thing name used by the OTA agent.
+ *
+ * AWS IoT requires Thing names to be unique for each device that connects to the broker.
+ * Likewise, the OTA agent requires the developer to construct and pass in the Thing name when
+ * initializing the OTA agent. The agent uses this size to allocate static storage for the
+ * Thing name used in all OTA base topics. Namely $aws/things/<thingName>
+ */
+#define otaconfigMAX_THINGNAME_LEN             64U
```
Comment (pvyawaha) [inline vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/ota_config.h:96]: Please set this to 128

**Review by pvyawaha — CHANGES_REQUESTED** [review]
Please fix the maximum thing name length in both the OTA conifgs.

File: vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/ota_config.h (hunk 6 above)
Comment (yanjos-dev) [reply vendors/ti/boards/cc3220_launchpad/aws_tests/config_files/ota_config.h]: done

File: vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_config.h (hunk 5 above)
Comment (yanjos-dev) [reply vendors/ti/boards/cc3220_launchpad/aws_demos/config_files/ota_config.h]: done

**Review by divekarshubham — APPROVED** [review]

**Review by pvyawaha — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**